src/main/resources/com/owlcyberdefense/syslog/xsd/syslog-rfc5424.dfdl.xsd

```diff
@@ -149,20 +149,11 @@ SOFTWARE.
   <xs:group name="rfc5424BodyGroup">
     <xs:sequence>
       <xs:choice>
-        <!--
-        NoStructuredData was an element with length="0" and an initiator="-"
-        but that's not portable as IBM DFDL doesn't put that into the infoset.
-        So we model this as a length 1 element instead.
-
-        <xs:element name="NoStructuredData" type="xs:string" dfdl:lengthKind="explicit" dfdl:length="1">
-          <xs:annotation>
-          <xs:appinfo source="http://www.ogf.org/dfdl/">
-            <dfdl:assert>{ . eq '-' }</dfdl:assert>
-          </xs:appinfo>
-          </xs:annotation>
+        <xs:element name="NoStructuredData" dfdl:initiator="-">
+          <xs:complexType>
+            <xs:sequence />
+          </xs:complexType>
         </xs:element>
```

Comment:
Actually, ignore this change. I tested with IBM DFDL and it does not allow complex types with empty content. Need another alternative that can work with Xerces-C. The commented out `NoStructureData` with length 1 and an assertion that the parsed value is a hyphen may be the best approach.

Reply:
Agree. 

Sticking with the IBM portable DFDL subset requires you to do this with a real element, string length 1 that must be a hyphen. 

I can't see any way to avoid it. 

You should use an explicit simple type so you can put an enumeration facet on it with value="-". That defends unparsing (so long as one validates before unparsing). It will also silence schema quality checkers like xsat2.